Source/SwiftLintFramework/Rules/Metrics/EnumCaseAssociatedValuesLengthRule.swift

```diff
@@ -0,0 +1,74 @@
+import SourceKittenFramework
+
+public struct EnumCaseAssociatedValuesLengthRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityLevelsConfiguration(warning: 5, error: 6)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "enum_case_associated_values_count",
+        name: "Enum Case Associated Values Count",
+        description: "Number of associated values should be low",
+        kind: .metrics,
+        nonTriggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, designation: String, contactNumber: Int)
+                case partTime(name: String, age: Int, contractEndDate: Date)
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, age: Int, designation: String, contactNumber: Int)
+                case partTime(name: String, contractEndDate: Date, age: Int, designation: String, contactNumber: Int)
+            }
+            """
+        ]
+    )
+
+    public func validate(
+        file: SwiftLintFile,
+        kind: SwiftDeclarationKind,
+        dictionary: SourceKittenDictionary
+    ) -> [StyleViolation] {
+        guard kind == .enum else {
```

Comment:
could you check for `enumElement` here and remove the substructure navigation below? this makes the rule more focused and leaves the navigation to `ASTRule`